firecracker.go

```diff
@@ -63,88 +56,5 @@ func NewFirecrackerClient(socketPath string, logger *logrus.Entry, debug bool) *
 
 	httpClient.SetTransport(transport)
 
-	return &FirecrackerClient{client: httpClient}
-}
-
-func (f *FirecrackerClient) PutLogger(ctx context.Context, logger *models.Logger) (*ops.PutLoggerNoContent, error) {
```

Comment:
So I get that this is a wrapper and it's going to be tedious to maintain, but are we really ready to get rid of it?  The wrapper does a couple of things that I think are valuable:

* For trivial operations (like PutLogger), it gets rid of the need to manage parameter structs like `PutLoggerParams`.  The result of this is a more natural Go API that involves function arguments rather than structs.
* For many of the operations, we set default timeout values.  Without something like this, every caller here would need to worry about the timeouts.

You can already see the effect this has on callers when you look at the changes that you had to make to `machine.go`.

Reply:
> For trivial operations (like PutLogger), it gets rid of the need to manage parameter structs like PutLoggerParams. The result of this is a more natural Go API that involves function arguments rather than structs.

Yea, but I don't know if that's worth keeping for a minor improvement in ease of use with now needing to maintain this. The goal of this PR was to eliminate any upkeep of structures when the swagger definition changes. And this could also be fixed to adding a factory function if needed.

> For many of the operations, we set default timeout values. Without something like this, every caller here would need to worry about the timeouts.

For the timeouts, I would be more inclined to agree to keep this if we were using those operation in multiple spots. However, since it is only being used in one location, I don't see any reason to keep it.  If there is a case where we need to call the operation in two spots that can be mitigated with helper functions.

Reply:
Actually I'll go ahead and do that, add a customization in the template to add a `PutLogger` customization.